src/pages/upload/index.tsx

```diff
@@ -74,88 +86,206 @@ const fileUploadReducer = (state: any, action: any) => {
+              value={courseInstructorId}
+              onChange={(e) => {
+                setCourseInstructorId(e.target.value)
+              }}
+            >
+              {instructors.map(
+                ({
+                  eggheadInstructorId,
+                  person,
+                }: {
+                  eggheadInstructorId: string
+                  person: {name: string}
+                }) => {
+                  return (
+                    <option value={eggheadInstructorId}>
+                      {person['name']}
+                    </option>
+                  )
+                },
+              )}
+            </select>
+          </div>
+          <div className="space-y-1">
+            <label className="block text-sm font-medium text-gray-700">
+              Video Files
+            </label>
+            {/* Drop zone UI adapted from https://larainfo.com/blogs/tailwind-css-drag-and-drop-file-upload-ui */}
+            <div className="max-w-xl">
+              <label className="flex justify-center w-full h-32 px-4 transition bg-white border-2 border-gray-300 border-dashed rounded-md appearance-none cursor-pointer hover:border-gray-400 focus:outline-none">
+                <span className="flex items-center space-x-2">
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    className="w-6 h-6 text-gray-600"
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    stroke="currentColor"
+                    stroke-width="2"
+                  >
+                    <path
+                      stroke-linecap="round"
+                      stroke-linejoin="round"
+                      d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12"
+                    />
+                  </svg>
+                  <span className="font-medium text-gray-600">
+                    Drop video files, or{' '}
+                    <span className="text-blue-600 underline">browse</span>
+                  </span>
+                </span>
+                <VideoUploader
+                  dispatch={dispatch}
+                  setLessonMetadata={setLessonMetadata}
+                />
+              </label>
+            </div>
+          </div>
+          {fileUploadState.files.map((file) => {
+            if (file.percent < 100) {
+              return (
+                <div>
+                  {file.file.name} - {file.message} - {file.percent}%
+                </div>
+              )
+            } else {
+              return null
+            }
+          })}
+          {values.lessons.map((lesson, i) => (
+            <div className="space-y-1">
+              <p className="block text-xs font-medium text-gray-600 uppercase">
+                Lesson ({i + 1}/{values.lessons.length})
+              </p>
```

Comment:
Once the videos have been uploaded to S3, you get inline forms for setting the lesson metadata (at the moment just _Title_).

Reply:
We could probably turn this div block into a component